xray/grpc.go

```diff
@@ -0,0 +1,143 @@
+package xray
+
+import (
+	"context"
+	"errors"
+	"github.com/golang/protobuf/proto"
```

Comment:
Can we format this file to avoid review dog warnings?

Reply:
Ran the `go fmt` for the file, those warning should be fixed.